src/sinks/aws_s_s/config.rs

```diff
@@ -102,68 +87,14 @@ pub enum Encoding {
     Json,
 }
 
-impl GenerateConfig for SqsSinkConfig {
-    fn generate_config() -> toml::Value {
-        toml::from_str(
-            r#"queue_url = "https://sqs.us-east-2.amazonaws.com/123456789012/MyQueue"
-            region = "us-east-2"
-            encoding.codec = "json""#,
-        )
-        .unwrap()
-    }
-}
-
-#[async_trait::async_trait]
-#[typetag::serde(name = "aws_sqs")]
-impl SinkConfig for SqsSinkConfig {
-    async fn build(
-        &self,
-        cx: SinkContext,
-    ) -> crate::Result<(crate::sinks::VectorSink, crate::sinks::Healthcheck)> {
-        let client = self.create_client(&cx.proxy).await?;
-        let healthcheck = self.clone().healthcheck(client.clone()).boxed();
-        let sink = super::sink::SqsSink::new(self.clone(), client)?;
-        Ok((
-            crate::sinks::VectorSink::from_event_streamsink(sink),
-            healthcheck,
-        ))
-    }
-
-    fn input(&self) -> Input {
-        Input::new(self.encoding.config().input_type() & DataType::Log)
-    }
-
-    fn acknowledgements(&self) -> &AcknowledgementsConfig {
-        &self.acknowledgements
-    }
+#[derive(Clone, Debug)]
+pub struct ConfigWithIds {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I think the implementation could be simplified a bit and render this struct not necessary.

This seems to exist only to provide a means for the request builder to call the message_group_id / message_deduplication_id helper functions.
I would suggest instead of having these be helper functions operating on a config struct, extract them as stand alone helper functions that just take the respective fields as input, and the fifo param can be a bool parameter as well.
Those can then be called in the specific `config.rs` files, in the SinkConfig implementations. That `build()` implementation is also where I would recommend instantiating the `RequestBuilder` , which can be passed into the `SSSink::new()` , along with the request (instead of passing the config which is just used to get those).

Reply:
done